frontend/app/components/collapsible.tsx

```diff
@@ -0,0 +1,48 @@
+import type { ComponentProps } from 'react';
+
+import { cn } from '~/utils/tw-utils';
+
+const variants = {
+  default: 'border border-gray-300 bg-gray-200 text-slate-700 hover:bg-neutral-300',
+};
+
+export interface CollapsableDetailsProps extends ComponentProps<'details'> {
+  id: string;
+  title: string;
+  required?: boolean;
+  label?: string;
+  variant?: keyof typeof variants;
+}
+
+export interface CollapsableSummaryProps extends ComponentProps<'details'> {
+  id: string;
+  title: string;
+  required?: boolean;
+  label?: string;
+  variant?: keyof typeof variants;
+}
```

Comment:
You want to extends `ComponentProps<'summary'>` and force `id` prop. You will use `children` prop from `summary` to provide the content.
```suggestion
export interface CollapsableSummaryProps extends ComponentProps<'summary'> {
  id: string;
}
```